src/workspaces/common/state/useWorkspaceStatePolling.ts

```diff
@@ -0,0 +1,121 @@
+import { LoadedState } from '@terra-ui-packages/core-utils';
+import _ from 'lodash/fp';
+import { useEffect, useRef, useState } from 'react';
+import { Ajax } from 'src/libs/ajax';
+import { workspacesStore, workspaceStore } from 'src/libs/state';
+import { pollWithCancellation } from 'src/libs/utils';
+import { BaseWorkspaceInfo, WorkspaceState, WorkspaceWrapper as Workspace } from 'src/workspaces/utils';
+
+// type WorkspaceUpdate = Partial<Workspace['workspace']>
+
+interface WorkspaceUpdate extends Partial<BaseWorkspaceInfo> {
+  workspaceId: string;
+  state: WorkspaceState;
+  errorMessage?: string;
+}
+
+const updateWorkspacesList = (workspaces: Workspace[], update: WorkspaceUpdate): Workspace[] =>
+  workspaces.map((ws) => updateWorkspaceIfMatching(ws, update));
+
+const updateWorkspaceIfMatching = <T extends Workspace>(workspace: T, update: WorkspaceUpdate): T => {
+  if (workspace.workspace?.workspaceId === update.workspaceId) {
+    const updatedWs = _.cloneDeep(workspace);
+    updatedWs.workspace.state = update.state;
+    updatedWs.workspace.errorMessage = update.errorMessage;
+    return updatedWs;
+  }
+  return workspace;
+};
+
+const checkWorkspaceState = async (workspace: Workspace, abort: () => void, signal: AbortSignal) => {
+  const doUpdate = (abort: () => void, update: WorkspaceUpdate) => {
+    workspacesStore.update((wsList) => updateWorkspacesList(wsList, update));
+    workspaceStore.update((ws) => {
+      if (ws) {
+        return updateWorkspaceIfMatching(ws, update);
+      }
+      return undefined;
+    });
+    abort();
```

Comment:
Hmm, I'm still not understanding the point of the abort.

Reply:
It's to cause the `pollWithCancellation` to stop.
Without the abort, you can have the polling loop continuing an extra cycle before react re-renders and calls the abort when cleaning up the old hook. 
Since the workspace it's polling on is no longer in a 'Deleting' state, it won't expect the 404, and the error will be re-thrown and surface as an error notification.

So basically, it's to ensure things stay in a consistent state.